toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/FeatureForm.kt

```diff
@@ -0,0 +1,334 @@
+import com.arcgismaps.mapping.featureforms.FieldFormElement
+import com.arcgismaps.mapping.featureforms.GroupFormElement
+import com.arcgismaps.mapping.featureforms.RadioButtonsFormInput
+import com.arcgismaps.mapping.featureforms.SwitchFormInput
+import com.arcgismaps.mapping.featureforms.TextAreaFormInput
+import com.arcgismaps.mapping.featureforms.TextBoxFormInput
+import com.arcgismaps.toolkit.featureforms.internal.components.base.BaseFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.base.BaseGroupState
+import com.arcgismaps.toolkit.featureforms.internal.components.base.FormStateCollection
+import com.arcgismaps.toolkit.featureforms.internal.components.base.MutableFormStateCollection
+import com.arcgismaps.toolkit.featureforms.internal.components.base.getState
+import com.arcgismaps.toolkit.featureforms.internal.components.base.rememberBaseGroupState
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.rememberComboBoxFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.rememberRadioButtonFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.codedvalue.rememberSwitchFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.datetime.rememberDateTimeFieldState
+import com.arcgismaps.toolkit.featureforms.internal.components.formelement.FieldElement
+import com.arcgismaps.toolkit.featureforms.internal.components.formelement.GroupElement
+import com.arcgismaps.toolkit.featureforms.internal.components.text.rememberFormTextFieldState
+import com.arcgismaps.toolkit.featureforms.internal.utils.FeatureFormDialog
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * The "property" determines the behavior of when the validation errors are visible.
+ */
+public sealed class ValidationErrorVisibility {
+
+    /**
+     * Indicates that the validation errors are only visible for editable fields that have
+     * received focus.
+     */
+    public object Automatic : ValidationErrorVisibility()
+
+    /**
+     * Indicates the validation is run for all the editable fields regardless of their focus state,
+     * and any errors are shown.
+     */
+    public object Visible : ValidationErrorVisibility()
+}
+
+/**
+ * A composable Form toolkit component that enables users to edit field values of features in a
+ * layer using forms that have been configured externally (using either in the the Web Map Viewer
+ * or the Fields Maps web app).
+ *
+ * @param featureForm The [FeatureForm] configuration.
+ * @param modifier The [Modifier] to be applied to layout corresponding to the content of this
+ * FeatureForm.
+ * @param validationErrorVisibility The [ValidationErrorVisibility] that determines the behavior of
+ * when the validation errors are visible. Default is [ValidationErrorVisibility.Automatic] which
+ * indicates errors are only visible once the respective field gains focus.
+ *
+ * @since 200.2.0
```

Comment:
```suggestion
 * @since 200.4.0
```
I think we should bump this up to 200.4